src/db_api_util.ts

```diff
@@ -86,6 +86,184 @@ export type DatabaseCommandEntry = {
   failureReason?: string | null;
 };
 
+export type DatabaseThreadEntry = {
+  slackParentMessageTs: string | null;
+  channelId: string | null;
+  userId: string | null;
+  userPhoneNumber: string | null;
+  needsAttention: boolean | null;
+};
+
+export type ThreadInfo = {
+  slackParentMessageTs: string;
+  channelId: string;
+  userId: string | null;
+  userPhoneNumber: string | null;
+  age: number | null;
+};
+
+export async function newThreadToDb(
+  databaseThreadEntry: DatabaseThreadEntry
+): Promise<void> {
+  const client = await pool.connect();
+  try {
+    await client.query(
+      'INSERT INTO threads (slack_parent_message_ts, channel_id, user_id, user_phone_number, needs_attention, updated_at) VALUES ($1, $2, $3, $4, $5, NOW())',
+      [
+        databaseThreadEntry.slackParentMessageTs,
+        databaseThreadEntry.channelId,
+        databaseThreadEntry.userId,
+        databaseThreadEntry.userPhoneNumber,
+        databaseThreadEntry.needsAttention,
+      ]
+    );
+    logger.info('DBAPIUTIL.newThreadToDb: Successfully created thread');
+  } catch (error) {
+    logger.info('Failed to update threads; ignoring for now!');
+  } finally {
+    client.release();
+  }
+}
+
+export async function setThreadNeedsAttentionToDb(
+  slackParentMessageTs: string,
+  needsAttention: boolean
+): Promise<void> {
+  const client = await pool.connect();
+  try {
+    await client.query(
+      'UPDATE threads SET needs_attention = $1 WHERE slack_parent_message_ts = $2;',
+      [needsAttention, slackParentMessageTs]
+    );
+    logger.info(
+      `DBAPIUTIL.setThreadNeedsAttentionToDb: Set thread ${slackParentMessageTs} needs_attention=${needsAttention}`
+    );
+  } catch (error) {
+    logger.info('Failed to update threads; ignoring for now!');
+  } finally {
+    client.release();
+  }
+}
+
+export async function getThreadLatestMessage(
+  slackParentMessageTs: string
+): Promise<string | null> {
+  const client = await pool.connect();
+  try {
+    const result = await client.query(
+      `SELECT slack_message_ts FROM messages
+      WHERE slack_parent_message_ts=$1 AND slack_message_ts IS NOT NULL
+      ORDER BY COALESCE(slack_send_timestamp, slack_receive_timestamp) DESC LIMIT 1`,
+      [slackParentMessageTs]
+    );
+    if (result.rows.length > 0) {
+      return result.rows[0]['slack_message_ts'];
+    }
+    return null;
+  } finally {
+    client.release();
+  }
+}
+
+export async function getUnclaimedVoters(
+  channelId: string | null
+): Promise<ThreadInfo[]> {
+  const client = await pool.connect();
+  try {
+    let result = null;
+    if (channelId) {
+      result = await client.query(
+        `SELECT
+         t.slack_parent_message_ts as id, t.channel_id, t.user_phone_number, t.user_id, EXTRACT(EPOCH FROM now() - t.updated_at) as age
+        FROM threads t
+        WHERE t.needs_attention AND t.channel_id = $1
+          AND NOT EXISTS (SELECT FROM volunteer_voter_claims c WHERE t.user_id=c.user_id
```

Comment:
you shouldn't need to check both the user_id and user_phone_number -- the user_id is derived from the user_phone_number